eris.go

```diff
@@ -95,6 +96,19 @@ func Errorf(format string, args ...any) statusError {
 	}
 }
 
+type joinError interface {
+	Unwrap() []error
+}
+
+// Join returns an error that wraps the given errors.
+func Join(errs ...error) error {
+	internal := errors.Join(errs...)
```

Comment:
This will break `eris.Is`'s behavior, which I'm not so sure.

Caller can be easily mistaken and make the wrong assumption.

Reply:
I think we should for sure test that `eris.Is` still works with this implementation 

First simple test in https://github.com/risingwavelabs/eris/pull/28 shows that it still seems to work. In the end we inherit the behaviour of [errors.Join](https://pkg.go.dev/errors#Join)

I did not want to push into your branch, that is why I created the tiny PR @Gogomoe

Reply:
Sorry i forget to reply the comment. The eris.is and as are implemented in #27. I also added more testcases to cover the join error